src/solvers/cholmod.jl

```diff
@@ -151,7 +151,18 @@ macro cholmod_param(kwarg, code)
     end
 end
 
-const COMMONS = Vector{Ref{cholmod_common}}()
+function getcommon()
+    if !haskey(task_local_storage(), :cholmod_common) # havent yet started cholmod on this task
+        common = finalizer(cholmod_l_finish, Ref(cholmod_common()))
+        result = cholmod_l_start(common)
+        @assert result == TRUE "failed to run `cholmod_l_start`!"
+        common[].print = 0  # no printing from CHOLMOD by default
+        common[].error_handler = @cfunction(error_handler, Cvoid, (Cint, Cstring, Cint, Cstring))
+        task_local_storage(:cholmod_common, common)
+    else
+        return task_local_storage(:cholmod_common)
```

Comment:
TLS is untyped so you might want to add a `::T` here. Also I think it supports the `get!` interface